plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/TrinoParquetDataSource.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.redshift;
+
+import io.airlift.slice.Slice;
+import io.trino.filesystem.TrinoInput;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.parquet.AbstractParquetDataSource;
+import io.trino.parquet.ParquetDataSourceId;
+import io.trino.parquet.ParquetReaderOptions;
+
+import java.io.IOException;
+
+public class TrinoParquetDataSource
```

Comment:
You can move `io.trino.plugin.hive.parquet.TrinoParquetDataSource` to `io.trino.parquet` package and `io.trino.plugin.hive.FileFormatDataSourceStats` to `io.trino.plugin.base.metrics` package in a preparatory commit and use that, no need to copy this.

Reply:
I am deferring this for now as it requires additional changes

Reply:
I moved FileFormatDataSourceStats but deferring moving TrinoParquetDataSource for now as it requires additional changes.